graphrag/index/context_switch/contextSwitcher.py

```diff
@@ -1,22 +1,247 @@
 from graphrag.common.progress import ProgressReporter
 from graphrag.config import GraphRagConfig
+from graphrag.config.enums import StorageType
+from graphrag.common.storage import PipelineStorage, BlobPipelineStorage, FilePipelineStorage
+from graphrag.common.utils.context_utils import get_files_by_contextid
+import pandas as pd
+from typing import cast
+from azure.core.exceptions import ResourceNotFoundError
+import asyncio
+from io import BytesIO
+from pathlib import Path
+from graphrag.config import (
+    create_graphrag_config,
+    GraphRagConfig,
+)
+from common.graph_db_client import GraphDBClient
+import os
+from graphrag.vector_stores import VectorStoreFactory, VectorStoreType
+from graphrag.vector_stores.base import BaseVectorStore
+from graphrag.vector_stores.lancedb import LanceDBVectorStore
+from graphrag.vector_stores.kusto import KustoVectorStore
+from graphrag.query.indexer_adapters import (
+    read_indexer_covariates,
+    read_indexer_entities,
+    read_indexer_relationships,
+    read_indexer_reports,
+    kt_read_indexer_reports,
+    read_indexer_text_units,
+)
+from graphrag.model.entity import Entity
 
 class ContextSwitcher:
     """ContextSwitcher class definition."""
 
-    def __init__(self):
-        #initialize Gremline and Cosmos Db client here.
-        pass
-    def activate(self, config: GraphRagConfig | str, contextId: str | None, reporter: ProgressReporter):
+    def __init__(self, root_dir:str , config_dir:str,reporter: ProgressReporter,
+                 context_id:str, community_level:int ,
+                 data_dir: str = None,
+                 optimized_search: bool= False):
+
+        self.root_dir=root_dir
+        self.config_dir=config_dir
+        self.data_dir=data_dir
+        self.reporter=reporter
+        self.context_id=context_id
```

Comment:
Wouldn't it make more sense to have the context_id be part of activation? This is a new ContextSwitcher per context